watt.py

```diff
@@ -173,23 +176,19 @@ def download_gcs_blob_to_tmp_file(self, blob_path: str, tmp_file_name: str) -> N
         blob = bucket.blob(blob_str)
         blob.download_to_filename(tmp_file_name)
 
-    @staticmethod
-    def read_in_chunks(file_object, chunk_size=1024*1024):
+    def read_in_chunks(self, file_object, line_skip_regex: re.Pattern):
         while True:
-            data = file_object.readlines(chunk_size)
+            data = file_object.readlines(self.chunk_size)
             if not data:
                 break
-            yield data
-
-    @staticmethod
-    def compare_files(file1: io.TextIOWrapper, file2: io.TextIOWrapper, line_skip_regex_str: str = None) -> int:
-        line_skip_regex = None
-        if line_skip_regex_str:
-            line_skip_regex = re.compile(line_skip_regex_str)
-        for chunk1, chunk2 in zip(CompareOutputs.read_in_chunks(file1), CompareOutputs.read_in_chunks(file2)):
-            if line_skip_regex:
-                chunk1 = [line for line in chunk1 if not line_skip_regex.search(line)]
-                chunk2 = [line for line in chunk2 if not line_skip_regex.search(line)]
+            for line in data:
+                if line_skip_regex and line_skip_regex.search(line):
+                    continue
+                yield line
+
+    def compare_files(self, file1: io.TextIOWrapper, file2: io.TextIOWrapper, line_skip_regex_str: str = None) -> int:
+        line_skip_regex = re.compile(line_skip_regex_str) if line_skip_regex_str else None
+        for chunk1, chunk2 in zip(self.read_in_chunks(file1, line_skip_regex), self.read_in_chunks(file2, line_skip_regex)):
```

Comment:
I think these local variables should be `line1` and `line2`, right? The generator returns the lines which don't match the regex within the loaded chunk. This way there's no possible misalignment of filtered chunk sizes for the comparison

Reply:
good call, just changed.